modules/payments/backend/modules/payments/viewsets.py

```diff
@@ -60,3 +69,41 @@ def get(self, request, *args, **kwargs):
             "data": history
         }
         return Response(response, status=status.HTTP_200_OK)
+
+class AppleIAProductsView(APIView):
+    authentication_classes = [authentication.TokenAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get(self, request, *args, **kwargs):
+        products = AppleIAPProduct.objects.filter(is_active=True)
+        response = {
+            "success": True,
+            "data": [obj.as_dict() for obj in products]
+        }
+        return Response(response, status=status.HTTP_200_OK)
+
+
+class AppleIAPayment(ViewSet):
+    serializer_class = appleIAPSerializer
+    authentication_classes = [authentication.TokenAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+
+    @csrf_exempt
+    def create(self, request):
+        data_copy = request.data.copy()
+        data_copy['user'] = request.user
+        serializer = self.serializer_class(
+            data=data_copy, context={"request": request}
+        )
+        data = None
+        if serializer.is_valid(raise_exception=True):
+            verify_receipt, success = ApplePaymentService.verify_apple_receipt(request.data)
+            print('verify_receipt', verify_receipt)
+            if success:
+                data = "success"
+            else:
+                data = "fail"
```

Comment:
Should the failure case be a different response status code? Perhaps you could do one `Response` for each case here.